synapse/api/errors.py

```diff
@@ -100,6 +100,12 @@ class Codes(str, Enum):
 
     UNREDACTED_CONTENT_DELETED = "FI.MAU.MSC2815_UNREDACTED_CONTENT_DELETED"
 
+    # Returned for federation requests where we can't process a request as we
+    # can't ensure the sending server is in a room which is partial-stated on
+    # our side.
+    # Part of MSC3706-rei1.
+    UNABLE_DUE_TO_PARTIAL_STATE = "ORG.MATRIX.MSC3706_UNABLE_DUE_TO_PARTIAL_STATE"
+
```

Comment:
What are the intended semantics here---how should a HS behave if they receive this error code? Is it "try again later", or "try asking someone else" maybe?

Put differently: I'm not sure if the error code adds much over a generic HTTP 503 response? But that's not to say we shouldn't include it for diagnostics and debugging.

Reply:
I do like it for diagnostics if nothing else; makes it very clear what's going on.

But I also think it might have value for explaining the reason why a request is being failed and then letting homeservers choose to backoff differently (the server isn't down, it's just not in the room yet, but you can usually expect it to join eventually).